src/ring/middleware/authorization.clj

```diff
@@ -0,0 +1,84 @@
+(ns ring.middleware.authorization
+  (:refer-clojure :exclude [update-keys])
+  (:require
+    [clojure.string :as str]
+    [ring.util.parsing :as parsing]))
+
+;; Copied from Clojure 1.11:
+(defn- update-keys
+  [m f]
+  (let [ret (persistent!
+              (reduce-kv (fn [acc k v] (assoc! acc (f k) v))
+                         (transient {})
+                         m))]
+    (with-meta ret (meta m))))
+
+(def ^{:added "0.4"} re-token68
+  "Pattern for base64/base64url/base32/base16 tokens.
+
+  See RFC 7235 Section 2 (https://datatracker.ietf.org/doc/html/rfc7235#section-2),
+  and RFC 9110 Section 11.2 (https://datatracker.ietf.org/doc/html/rfc9110#section-11.2)."
+  (re-pattern "[A-Za-z0-9._~+/-]+=*"))
+
+(def ^{:added "0.4"} re-auth-param
+  "Pattern for authentication parameters.
+
+  See RFC 7235 Section 2 (https://datatracker.ietf.org/doc/html/rfc7235#section-2),
+  and RFC 9110 Section 11.2 (https://datatracker.ietf.org/doc/html/rfc9110#section-11.2)."
+  (re-pattern (str "(" parsing/re-token ")\\s*=\\s*(?:" parsing/re-value ")")))
+
+(defn authorization
+  "Parse 'Authorization' header.
+
+  See RFC 7235 Section 2 (https://datatracker.ietf.org/doc/html/rfc7235#section-2),
+  and RFC 9110 Section 11 (https://datatracker.ietf.org/doc/html/rfc9110#section-11)."
+  {:added "0.4"}
+  [request]
+  (when-let [[auth-scheme token-or-params] (some-> (get-in request [:headers "authorization"])
+                                                   (str/split #"\s" 2))]
+    (cond
+      (empty? token-or-params)
+      {:scheme auth-scheme}
+
+      (re-matches re-token68 token-or-params)
+      {:scheme  auth-scheme
+       :token68 token-or-params}
+
+      :else
+      (let [auth-params (->> (str/split token-or-params #"\s*,\s*")
+                             (map #(re-matches re-auth-param %))
+                             (filter some?)
+                             (map (fn [[_ k v1 v2]]
+                                    [k (or v1 v2)]))
+                             (into {}))]
+        {:scheme auth-scheme
+         :params auth-params}))))
+
+(defn authorization-request
+  "Parses `Authorization` header in the request map. See: wrap-authorization.
+
+  See RFC 7235 Section 2 (https://datatracker.ietf.org/doc/html/rfc7235#section-2),
+  and RFC 9110 Section 11 (https://datatracker.ietf.org/doc/html/rfc9110#section-11)."
+  {:added "0.4"}
+  [request]
+  (if (:authorization request)
+    request
+    (assoc request :authorization (some-> (authorization request)
+                                          (update :scheme str/lower-case)
+                                          (update :params update-keys str/lower-case)))))
```

Comment:
Ensure line length is within 80 characters. (Looks like this repo doesn't have the Ring contributing guidelines linked, so I'll need to add that!)